src/pools/abstract-pool.ts

```diff
@@ -399,4 +415,10 @@ export abstract class AbstractPool<
       }
     }
   }
+
+  private checkEmitBusy (): void {
```

Comment:
I don't know why, but I don't like the function's name (`checkEmitBusy`) :confused:  
Maybe `checkAndEmitBusy`? Other suggestions?